File: 2trabajoConstructores/c#/claseAbstracta/AnimalSobrecargado.cs
public class AnimalSobrecargado : AnimalBase
{
    public AnimalSobrecargado(string nombre)
    {
        Nombre = nombre;
        Sonido = "N/A";
    }

    public AnimalSobrecargado(string nombre, string sonido)
    {
        Nombre = nombre;
        Sonido = sonido;
    }

    public override void HacerSonido()
    {
        Console.WriteLine($"{Nombre} hace: {Sonido}");
    }
}

File: 2trabajoConstructores/js/clase abstracta/AnimalSobrecargado.js
const Animal = require("./Animal.js");

class AnimalSobrecargado extends Animal {
    constructor(nombre, sonido) {
        super();
        this.nombre = nombre || "Animal por defecto";
        this.sonido = sonido || "Sonido por defecto";
    }

    hacerSonido() {
        console.log(`${this.nombre} hace: ${this.sonido}`);
    }
}

module.exports = AnimalSobrecargado;
